va/sysdeps.h

```diff
@@ -48,7 +48,7 @@
 # include <log/log.h>
 #endif
 
-#if defined __GNUC__ && defined HAVE_GNUC_VISIBILITY_ATTRIBUTE
+#if defined __GNUC__ && defined HAVE_GNUC_VISIBILITY_ATTRIBUTE && !defined(_WIN32)
```

Comment:
Hm. Actually we had this:
```
meson.build:125:  va_c_args += ['-DHAVE_GNUC_VISIBILITY_ATTRIBUTE']

# Symbol visibility
if cc.has_argument('-fvisibility=hidden')
  va_c_args += ['-DHAVE_GNUC_VISIBILITY_ATTRIBUTE']
endif
```

Why this did not do the job?

Reply:
The  `HAVE_GNUC_VISIBILITY_ATTRIBUTE` condition evaluates to false correctly for MSVC with `CC=cl`, but for MinGW with `CC=gcc` it evaluates to true, which causes these warnings

![image](https://user-images.githubusercontent.com/82116493/197647325-523b547b-982a-4913-98e3-c9e087db131f.png)

Reply:
So, we stepped into non 100% guarantee that has_argument() works correctly. I suggest to do this on meson.build level instead of in source.

Reply:
Added the condition in meson.build, validated no longer repro in mingw local build with the change

Reply:
> This is wrong, please do not do this. Meson natively handles visibility by detecting if it's appropriate for the compiler, and there is no reason to avoid defining this attribute arbitrarily -- if you have GNUC, set it.

@eli-schwartz : if meson supports that, why are we getting the following warning: See above discussion, meson detected that visibility is supported, passed the flag and we get the warning. Can you suggest what was done incorrectly? 
```
../va/va_trace.c:947:1: warning: visibility attribute not supported in this configuration; ignored [-Wattributes]
```

Reply:
I think the problem is MINGW gcc specifically, so I've now disabled the attributes specifically in that case, I'm open to adding a more comprehensive check if it's available in meson. I tried both ways below in meson and both ways are supported when building in gcc/mingw but then the warning shows up.

```
if cc.has_function_attribute('visibility:hidden')
if cc.has_argument('-fvisibility=hidden')
```